### DISCUSSION TITLE
Redirects not working as expected in vercel.json (Nuxt)

### DISCUSSION THREAD

**Discussion by danspratling** [vercel/vercel#7627] [Help]
I've been having issues with my vercel.json config which means redirects aren't working as I would expect

Here's what I have in my `verel.json` file

```
{
  redirects: [
    // Catch a specific route and redirect it to another page
    {
      "source": "/blog/post",
      "has": [
        {
          "type": "host",
          "value": "sub.domain.com"
        }
      ],
      "destination": "https://www.newdomain.com/sub/articles/new-post-title/"
    },
    // Catch all other routes and redirect to the homepage
    {
      "source": "/(.*)",
      "has": [
        {
          "type": "host",
          "value": "sub.domain.com"
        }
      ],
      "destination": "https://www.newdomain.com/sub/"
    },
  ]
}
```

What actually happens when I use this config is that every page, including /blog/post is being captured by the second rule and redirecting to the homepage. I would have thought rules applied earlier would be used first if they match (like in other redirect files) but that doesn't seem to be the case.

I've tried using rewrites, but as far as I can tell they aren't being recognised at all by vercel when I deploy. I'm not sure if I'm missing a config elsewhere to enable them but I've not had any luck figuring this out.

I'm deploying a nuxtjs project

Any ideas what I'm missing?

**Comment by wiiiimm** [comment]
I have an example Nuxt.js project set up at https://github.com/vercel-support/75290-nuxtjs-redirects with the following `vercel.json`.

```
{
  "redirects": [
    {
      "source": "/blog/post",
      "has": [
        {
          "type": "host",
          "value": "75290-nuxtjs-redirects.vercel.app"
        }
      ],
      "destination": "https://vercel.com/blog/how-the-web-evolves",
      "permanent": false
    },
    {
      "source": "/(.*)",
      "has": [
        {
          "type": "host",
          "value": "75290-nuxtjs-redirects.vercel.app"
        }
      ],
      "destination": "https://www.theverge.com/",
      "permanent": false
    }
  ]
}
```

If you visit it on its _[other domain](https://75290-nuxtjs-redirects.vercel-support.app/)_, you will see the actual Nuxt.js project.

If you visit https://75290-nuxtjs-redirects.vercel.app/blog/post, you will be redirected to https://vercel.com/blog/how-the-web-evolves and for everything else under https://75290-nuxtjs-redirects.vercel.app/, you will be taken to https://www.theverge.com.

A few things for you to check:

1. ensure you are using `vercel.json` and not `verel.json`.
2. ensure you do not have an existing route in nuxt.js for `/blog/post` because 
> Note: The source property should NOT be a file because precendence is given to the filesystem prior to rewrites being applied. Instead, you should rename your static file or Serverless Function.

**Comment by nedaa-ryad** [comment]
i have a problem deploying my Django Web Application and the JSON file when named with vercel.json is with gray icon but when write any names else vercel works successfully like {} means normal... I was trying a lot but i got the same issues 404 Code; `Not-Found`

and this my vercel.json 
{
  "builds": [
    {
      "src": "project/wsgi.py",
      "use": "@vercel/python",
      "config": { "maxLambdaSize": "15mb", "runtime": "python3.9" }
    },
    {
      "src": "build_files.sh",
      "use": "@vercel/static-build",
      "config": { "distDir": "static" }
    }
  ],
  "routes": [
    {
      "src": "/static/(.*)",
      "dest": "/static/$1"
    },
    {
      "src": "/media/(.*)",
      "dest": "/media/$1"
    },
    {
      "src": "/(.*)",
      "dest": "project/wsgi.py"
    }
  ]
}
